app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsPresenter.kt

```diff
@@ -1,53 +1,62 @@
 package ar.com.wolox.android.example.ui.home.news
 
-import ar.com.wolox.android.example.model.New
+import android.util.Log
+import ar.com.wolox.android.example.model.NewData
+import ar.com.wolox.android.example.network.builder.networkRequest
+import ar.com.wolox.android.example.network.repository.NewRepository
 import ar.com.wolox.android.example.utils.UserSession
-import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import ar.com.wolox.wolmo.core.presenter.CoroutineBasePresenter
+import kotlinx.coroutines.launch
 import javax.inject.Inject
+import kotlin.collections.ArrayList
 
-class NewsPresenter @Inject constructor(private val userSession: UserSession) : BasePresenter<NewsView>() {
+class NewsPresenter @Inject constructor(
+    private val userSession: UserSession,
+    private val newRepository: NewRepository
+) : CoroutineBasePresenter<NewsView>() {
 
     override fun onViewAttached() {
         super.onViewAttached()
         fetchNews()
     }
 
     // The presenter stores the news retrieved from backend.
-    private var news: ArrayList<New> = arrayListOf()
+    private var news: ArrayList<NewData> = arrayListOf()
 
-    private fun fetchNews() { // In the next card, this method will hit the backend to retrieve the first list of news.
-        news = arrayListOf(
-                NEWS[0],
-                NEWS[1],
-                NEWS[2],
-                NEWS[3],
-                NEWS[4]
-        )
-        view?.showNews(news)
-    }
+    private var currentPage: Int = 1
+    private var totalPages: Int = 1
 
-    fun updateNews() { // In the next card, this method will hit the backend to retrieve more news, and put it them into the current list.
-        val newsFetched = arrayListOf<New>()
-        news.addAll(newsFetched)
-        // Sort news by date
-        val anyNews = newsFetched.isNotEmpty()
-        if (anyNews) view?.updateNews(news) else view?.showNoNewNewsAlert()
+    private fun fetchNews() = launch {
+        networkRequest(newRepository.getNews(currentPage)) {
+            onResponseSuccessful { it ->
+                totalPages = it!!.totalPages
+                news = it.page
+                news.sortByDescending { newDate -> newDate.date }
+                currentPage++
+            }
+            onResponseFailed { e, m -> Log.i("NewsRequest", "Request failed : $e - $m") } // TODO Maybe of failed credentials, send user to login.
```

Comment:
The user has to know about this error. Maybe you can show a toast instead of logging it.